### PR TITLE
Avoid making unnecessary copies

### PR DESCRIPTION
**Pull request by ShFil119**
To think about:
- anonymous union
It prevents making reasonable contructor.
https://github.com/rwengine/openrw/blob/3c83e7cf52676dd27924bffb98713c712b4ded32/rwengine/src/script/ScriptMachine.cpp#L47
- opcode 0219
For example calling element nr size()-1  isn't thread safe.
https://github.com/rwengine/openrw/blob/512ac6f5eed3f6a356aac6bad3f28d8936faa25d/rwengine/src/script/modules/GTA3ModuleImpl.inl#L6280

### CODE REVIEW

File: rwengine/src/render/OpenGLRenderer.hpp [hunk 1]
```diff
@@ -14,23 +14,41 @@ typedef uint64_t RenderKey;
 
 typedef std::uint32_t RenderIndex;
 
-struct VertexP3 {
+// just a handle
+struct VertexP {
```
Comment (danhedron) [inline rwengine/src/render/OpenGLRenderer.hpp]: What is this for?

File: rwengine/src/loaders/LoaderIFP.hpp [hunk 2]
```diff
@@ -16,6 +16,17 @@ struct AnimationKeyframe {
...
+
+    AnimationKeyframe(glm::quat _rotation, glm::vec3 _position, glm::vec3 _scale, float _starttime, int _id)
+        : rotation(_rotation)
+        , position(_position)
+        , scale(_scale)
+        , starttime(_starttime)
+        , id(_id) {
+    }
+
+    AnimationKeyframe() {
```
Comment (danhedron) [inline rwengine/src/loaders/LoaderIFP.hpp]: Is it safe to leave the members uninitialised? Where is this ctor used

Comment (ShFil119) [reply rwengine/src/loaders/LoaderIFP.hpp]: Here:
https://github.com/ShFil119/openrw/blob/0b11659c81b4ee3e8dcd813a84af6a4ff5bb3a7d/rwengine/src/loaders/LoaderIFP.cpp#L35
But few moment later it's overwritten here:
https://github.com/ShFil119/openrw/blob/0b11659c81b4ee3e8dcd813a84af6a4ff5bb3a7d/rwengine/src/loaders/LoaderIFP.cpp#L9
If you want, I can change.

File: rwengine/src/render/OpenGLRenderer.hpp (hunk 1 above)
Comment (ShFil119) [reply rwengine/src/render/OpenGLRenderer.hpp]: I was thinking it - "@todo normalize this to have the same interface as VertexP3" can mean there's possibility that in future one vector will store VertexP2 and VertexP3.

**Pull request merged**